Src/WitsmlExplorer.Frontend/components/ContextMenus/CopyComponentsToServer.tsx

```diff
@@ -1,71 +1,89 @@
-import { DeleteComponentsJob } from "../../models/jobs/deleteJobs";
-import ObjectReference from "../../models/jobs/objectReference";
-import { ReplaceComponentsJob } from "../../models/jobs/replaceComponentsJob";
-import LogCurveInfo from "../../models/logCurveInfo";
-import ObjectOnWellbore, {
-  toObjectReference
-} from "../../models/objectOnWellbore";
+import { ComponentType } from "../../models/componentType";
 import { Server } from "../../models/server";
-import AuthorizationService from "../../services/authorizationService";
-import ComponentService from "../../services/componentService";
-import JobService, { JobType } from "../../services/jobService";
-import ObjectService from "../../services/objectService";
-import { displayMissingObjectModal } from "../Modals/MissingObjectModals";
-import { displayReplaceModal } from "../Modals/ReplaceModal";
-import { menuItemText, pluralize } from "./ContextMenuUtils";
+import { menuItemText } from "./ContextMenuUtils";
 import NestedMenuItem from "./NestedMenuItem";
+import { copyComponentsToServer } from "./CopyComponentsToServerUtils";
+import CopyRangeModal, { CopyRangeModalProps } from "../Modals/CopyRangeModal";
+import OperationType from "../../contexts/operationType";
 
 export interface CopyComponentsToServerMenuItemProps {
   componentsToCopy: { uid: string }[];
   componentType: ComponentType;
+  withRange?: boolean;
+}
+
+interface ComponentWithRange {
+  uid: string;
+  minIndex: number | Date;
+  maxIndex: number | Date;
 }
 
 export const CopyComponentsToServerMenuItem = (
   props: CopyComponentsToServerMenuItemProps
 ): React.ReactElement => {
-  const { componentsToCopy, componentType } = props;
+  const { componentsToCopy, componentType, withRange } = props;
   const {
     navigationState: { selectedServer, selectedObject, servers }
   } = useContext(NavigationContext);
   const { dispatchOperation } = useContext(OperationContext);
+  const menuComponents = menuItemText("copy", componentType, componentsToCopy);
+  const menuText =
+    withRange === true
+      ? menuComponents + " with range to server"
+      : menuComponents + " to server";
 
   return (
     <NestedMenuItem
       key={"copyComponentToServer"}
-      label={`${menuItemText(
-        "copy",
-        componentType,
-        componentsToCopy
-      )} to server`}
+      label={menuText}
       disabled={componentsToCopy.length < 1}
     >
       {servers.map(
         (server: Server) =>
           server.id !== selectedServer.id && (
             <MenuItem
               key={server.name}
-              onClick={() =>
-                copyComponentsToServer({
-                  targetServer: server,
-                  sourceServer: selectedServer,
-                  componentsToCopy,
-                  dispatchOperation,
-                  sourceParent: selectedObject,
-                  componentType
-                })
-              }
+              onClick={() => {
```

Comment:
Can you move the onClick logic into its own function to make it a bit more readable?

Reply:
solved